### PR TITLE
Nom 5/feature/repealing rules

### PR DESCRIPTION
**Pull request by AdamLucka**

### CODE REVIEW

**Review by Kirbitz — APPROVED** [review]
Looks good let's take down another story

**Review by meyert8034 — APPROVED** [review]
Looks good and all tests passed

File: src/main/kotlin/nomic/data/repositories/repealrule/RepealRuleRepository.kt [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            where {
+                it.ruleId eq ruleId
+            }
+        }
+        return db.update(Rules) {
+            set(it.active, false)
+            where {
+                it.ruleId eq ruleId
+            }
+        }
```
[the remark this answers is not in the document]
Comment (AdamLucka) [reply src/main/kotlin/nomic/data/repositories/repealrule/RepealRuleRepository.kt:32]: Does it matter how many rows are affected though? We just want to know that something was affected to know that a rule was updated.

File: src/main/kotlin/nomic/domain/repealrule/RepealRuleDomain.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * @see [nomic.domain.repealrule.IRepealRuleDomain]
+ * @see [nomic.data.repositories.repealrule.RepealRuleRepository]
+ * @param repealRuleRepository the instance of [RepealRuleRepository][nomic.data.repositories.repealrule.RepealRuleRepository]
+ * to use as a data collector
+ */
+@Service
+class RepealRuleDomain(private val repealRuleRepository: RepealRuleRepository) : IRepealRuleDomain {
+
+    override fun repealRule(ruleId: String): RepealRuleResponse {
+        val ruleIdInt: Int = ruleId.toIntOrNull() ?: throw IllegalArgumentException("Please enter a valid ruleId!")
```
[the remark this answers is not in the document]
Comment (AdamLucka) [reply src/main/kotlin/nomic/domain/repealrule/RepealRuleDomain.kt:21]: Because the endpoint returns a string and here is where we verify it is an integer

**Pull request merged**